src/quo/components/inputs/address_input/component_spec.cljs

```diff
@@ -38,47 +40,56 @@
                    {:scanned-value  scanned-value
                     :on-change-text on-change-text
                     :ens-regex      ens-regex}])
-        (h/wait-for #(h/is-truthy (h/get-by-label-text :clear-button)))
-        (h/was-called-with on-change-text scanned-value)
-        (h/has-prop (h/get-by-label-text :address-text-input) :default-value scanned-value))))
+        (-> (h/wait-for #(h/get-by-label-text :clear-button))
+            (.then (fn []
+                     (h/was-called-with on-change-text scanned-value)
+                     (h/has-prop (h/get-by-label-text :address-text-input)
+                                 :default-value
+                                 scanned-value)))))))
 
   (h/test "clear icon is shown when input has text"
     (with-redefs [clipboard/get-string #(% "")]
       (h/render [address-input/address-input
                  {:scanned-value "scanned value"
                   :ens-regex     ens-regex}])
-      (h/wait-for #(h/is-truthy (h/get-by-label-text :clear-button-container)))
-      (h/wait-for #(h/is-truthy (h/get-by-label-text :clear-button)))))
+      (-> (h/wait-for #(h/get-by-label-text :clear-button-container))
+          (.then #(h/is-truthy (h/get-by-label-text :clear-button))))))
 
   (h/test "on blur with text and blur? false"
     (with-redefs [clipboard/get-string #(% "")]
       (h/render [address-input/address-input
                  {:scanned-value "scanned value"
                   :ens-regex     ens-regex}])
-      (h/wait-for #(h/is-truthy (h/get-by-label-text :clear-button)))
-      (h/fire-event :on-focus (h/get-by-label-text :address-text-input))
-      (h/fire-event :on-blur (h/get-by-label-text :address-text-input))
-      (h/has-prop (h/get-by-label-text :address-text-input) :placeholder-text-color colors/neutral-30)))
+      (-> (h/wait-for #(h/get-by-label-text :clear-button))
```

Comment:
FYI: Notice how it's unnecessary to surround `get-by-*` with `h/is-truthy` when used as a matcher for `h/wait-for. The spec for RNTL `waitFor` clearly says the predicate should throw an exception when not found and every `get-by-*` function does that. Since this is a common need, it's nice to do less.